File: php/features.php
<?php
declare(strict_types=1);

// Cvedb SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CvedbFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CvedbBaseFeature();
            case "test":
                return new CvedbTestFeature();
            default:
                return new CvedbBaseFeature();
        }
    }
}

File: py/features.py
# Cvedb SDK feature factory

from feature.base_feature import CvedbBaseFeature
from feature.test_feature import CvedbTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CvedbBaseFeature(),
        "test": lambda: CvedbTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
